packages/liveblocks-react/src/comments/CommentsRoom.ts

```diff
@@ -100,10 +104,10 @@ export type ThreadsStateLoading = {
   error?: never;
 };
 
-export type ThreadsStateError = {
+export type ThreadsStateSuccessOrError<TThreadMetadata extends BaseMetadata> = {
   isLoading: false;
-  threads?: never;
-  error: Error;
+  threads: ThreadData<TThreadMetadata>[];
+  error?: Error;
 };
```

Comment:
Can there be a state with both `threads` and `error` now? If not, we should keep the error and success states separate since they can be used for narrowing (e.g. check if `threads` exist, if so, `error` will be typed as `never` as expected)

Reply:
That is correct. `threads` and `error` can both exist at the same time. I think this could provide better user experience to the end users - e.g., by keeping the existing threads data, users are not left with an empty or broken page if there's an error during revalidation. This ensures that the user experience remains uninterrupted. Also, since we have an error retry mechanism in place (and revalidation on browser 'online' events), and say network connection is lost only briefly, there won't be a drastic shift in the threads list.

Did you have a particular idea in mind with threads and error being mutually exclusive? @marcbouchenoire

Reply:
I just thought we didn't have that yet but I agree it's better this way 👍 I do think we could name the type something else, `ThreadsStateResolved` maybe.

Reply:
`ThreadsStateResolved` does sound more representative! I've added a small comment before the type to explain the type in slightly more detail.